### ISSUE TITLE
When a volume is a file, docker will delete its content when starting.

### ISSUE THREAD

**Issue by lazybean** [docker/for-win#8439]
- [x] I have tried with the latest version of my channel (Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 742EB8D3-9DAD-4542-8687-0F8B2BDEC3D1/20200911082441

### Expected behavior
The mounted file should stay intact
### Actual behavior
File content is erased.
### Information

  - Windows Version: 2004 (build 19041.508)
  - Docker Desktop Version: 2.3.6.2 (47842)
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: no

### Steps to reproduce the behavior
1. In a folder (dockerBug) create 2 files:

- foo.txt :  "hello docker"
- docker-compose.yml:
```
version: "3"

services:
  web:
    image: node:lts-alpine
    working_dir: /temp
    volumes:
      - ./foo.txt:/temp/foo.txt
```
1.b `git init .; git commit -a'
2. docker-compose up -d
3. restart docker
4. Notice foo.txt is empty
```
diff --git a/foo.txt b/foo.txt
index 8260792..e69de29 100644
--- a/foo.txt
+++ b/foo.txt
@@ -1 +0,0 @@
-hello docker
```

5. 
```
❯ git restore .
error: unable to unlink old 'foo.txt': Device or resource busy
```

**Comment by karasq** [comment]
Same problem. Since 2.3.5.0 every update is somehow broken. Since 2.3.6.1 files disappear (this version actually removed whole my mapped directory content) i guess 2.3.6.2 supposed to fix it, but mapped files still seems to be earsed after docker restart.

**Comment by simonferquel** [comment]
All right, we fixed the problem for directories on 2.3.6.2 but there is still an issue with single-file mounts. Fix is in the works.

**Comment by TBG-FR** [comment]
I was turning crazy with my volumes directories being deleted at each stop/start... I'll update to 2.3.6.2, but it looks like this fix isn't mentionned in the changelog...

**Comment by simonferquel** [comment]
Reopening (we merged a PR internally that triggered closing the issue, but the fix is not published yet)

**Comment by djs55** [comment]
I believe the fix is in the latest Docker Desktop Edge 2.3.7.0 (48173): https://docs.docker.com/docker-for-windows/edge-release-notes/#docker-desktop-community-2370 I'll close this ticket but let us know how it goes and feel free to open new tickets if you encounter more problems. Thanks for your reports!

**Comment by alister** [comment]
I'm running Docker Desktop for Windows, v2.4.1.0 (48583), edge.

The project I'm working on also has files mounted individually mounted as volumes into a container (hosted inside WSL2, Ubuntu 20.04), and if when they were edited, I got the "error: unable to unlink old 'filename': Device or resource busy" for each of the files (there are 3), until Docker for Desktop is shutdown, when I could change the files.  

Since the files had been edited on a Git branch, the 'Device or resource busy' made it impossible to change branches (till I shutdown Docker for Desktop).
